.github/workflows/pre-commit-autoupdate.yml

```diff
@@ -1,58 +1,69 @@
 
 jobs:
-  # This workflow contains a single job called "autoupdate"
+  # This workflow contains a single job called 'autoupdate'.
   autoupdate:
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v2
+      - name: Check out the repository
+        # Don't use release tags for actions:  https://julienrenaux.fr/2019/12/20/github-actions-security-risk/
+        # Commit is equivalent to release tag v2.3.2.
+        uses: actions/checkout@2036a08e25fa78bbd946711a407b529a0a1204bf
 
       - name: Set the name of the branch for these changes
         id: branch
         run: echo "::set-output name=branchname::pre-commit-autoupdate"
 
       - name: Set up Python
-        uses: actions/setup-python@v2
+        # Commit is equivalent to release tag v2.1.2.
+        uses: actions/setup-python@24156c231c5e9d581bde27d0cdbb72715060ea51
         with:
-          python-version: >3
+          python-version: '>=3'
 
       - name: Install pre-commit
         run: pip install pre-commit
 
-      - name: ${{ github.action }}
+      - name: ${{ github.workflow }}
         run: |
           pre-commit autoupdate
-          pre-commit run --all-files
-
-      - name: Create a new branch pre-commit-autoupdate and push the changes to it
-        run: |
-          git checkout -B ${{ steps.branch.branchname }}
-          git add .
-          git commit -m "${{ github.action }}"
-          git push -fu origin ${{ steps.branch.branchname }}
+          until pre-commit run --all-files
+          do
+            printf "\nRe-running pre-commits...\n"
+          done
 
       - name: Create pull request
         id: pr
-        uses: peter-evans/create-pull-request@v3
+        # Commit is equivalent to release tag v3.2.0.
+        uses: peter-evans/create-pull-request@9bf4b302a561e1fe9120f6dc81cc39daed984a99
         with:
-          branch: pre-commit-autoupdate
-          base: master
-          commit-message: ${{ github.action }}
-          title: ${{ github.action }}
+          branch: ${{ steps.branch.outputs.branchname }}
+          commit-message: ${{ github.workflow }}
+          title: ${{ github.workflow }}
+          body: Automated changes performed by a GitHub workflow.
 
       - name: Set the file name for the Towncrier news fragment
         id: newsfile
+        # Only run if the PR exists.
+        if: ${{ steps.pr.outputs.pull-request-number }}
         run: echo "::set-output name=filename::${{ steps.pr.outputs.pull-request-number }}.misc"
 
       - name: Make a Towncrier news fragment and push it to the branch
+        # Only run if the PR exists.
+        if: ${{ steps.newsfile.outputs.filename }}
         run: |
-          echo "${{ github.action }}" > ${{ steps.newsfile.filename }}
-          git add ${{ steps.newsfile.filename }}
+          echo "${{ github.workflow }}." > ${{ steps.newsfile.outputs.filename }}
+          git config user.name "GitHub"
+          git config user.email "noreply@github.com"
```

Comment:
I'm surprised that this is necessary

Reply:
I'll test it without.  I ran into trouble by not including it before, but the rest of the config has changed since and it may no longer be a problem.

Reply:
Yep, it's necessary:
```
  Make a Towncrier news fragment and push it to the branch0s
    PULL_REQUEST_NUMBER: 1
Run echo "Automatically update and run pre-commit hooks." > newsfragments/1.misc
Already on 'pre-commit-autoupdate'
Your branch is up to date with 'origin/pre-commit-autoupdate'.

*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: empty ident name (for <runner@fv-az29.jca0jqzscppeni1fo0mhnahhcf.gx.internal.cloudapp.net>) not allowed
##[error]Process completed with exit code 128.
```